src/formatter.ts

```diff
@@ -771,15 +768,38 @@ export class Formatter {
       lastContext.getMetrics().notePx -
       lastContext.getMetrics().totalRightPx -
       firstContext.getMetrics().totalLeftPx;
-    let targetWidth = adjustedJustifyWidth;
-    let actualWidth = shiftToIdealDistances(calculateIdealDistances(targetWidth));
     const musicFont = Flow.DEFAULT_FONT_STACK[0];
-    const paddingMax = musicFont.lookupMetric('stave.endPaddingMax');
-    const paddingMin = musicFont.lookupMetric('stave.endPaddingMin');
+    const configMinPadding = musicFont.lookupMetric('stave.endPaddingMin');
+    const configMaxPadding = musicFont.lookupMetric('stave.endPaddingMax');
+    let targetWidth = adjustedJustifyWidth;
+    const distances = calculateIdealDistances(targetWidth);
+    let actualWidth = shiftToIdealDistances(distances);
+    // Calculate right justification by finding max of (configured value, min distance between tickables)
+    // so measures with lots of white space use it evenly, and crowded measures use at least the configured
+    // space
+    const calcMinDistance = (targetWidth: number, distances: Distance[]) => {
+      let mdCalc = targetWidth / 2;
+      if (distances.length > 1) {
+        for (let di = 1; di < distances.length; ++di) {
+          mdCalc = Math.min(distances[di].expectedDistance / 2, mdCalc);
+        }
+      }
+      return mdCalc;
+    };
+    const minDistance = calcMinDistance(targetWidth, distances);
+    const multiNote = contextList.length > 1;
+    // right justify to either the configured padding, or the min distance between notes, whichever is greatest.
+    // This * 2 keeps the existing formatting unless there is 'a lot' of extra whitespace, which won't break
+    // existing visual regression tests.
+    const paddingMax = configMaxPadding * 2 < minDistance ? minDistance : configMaxPadding;
+    const paddingMin = paddingMax - (configMaxPadding - configMinPadding);
     const maxX = adjustedJustifyWidth - paddingMin;
 
     let iterations = maxIterations;
-    while ((actualWidth > maxX && iterations > 0) || (actualWidth + paddingMax < maxX && iterations > 1)) {
+    while (
```

Comment:
This is hard to read. Let's actually break this up:

```typescript
if (multinote && iterations > 0 && actualWidth > maxX) {
    while (actualWidth + paddingMax < maxX && iterations > 1) {
      ...
    }
}
```

I think that's equivalent, no?

Reply:
Okay, yeah, that's right. I think this would be right?

```typescript
if (multinote) {
    while ((actualWidth > maxX && iterations > 0) ||
           (actualWidth + paddingMax < maxX && iterations > 1)) {
      ...
    }
}
```